app/javascript/components/visualization/StudyViolinPlot.js

```diff
@@ -72,10 +73,28 @@ function RawStudyViolinPlot({
       if (!distributionPlotToUse) {
         distributionPlotToUse = defaultDistributionPlot
       }
+
+      const start = `perfTimeFrontendStart-${graphElementId}`
+      performance.mark(start)
+
       renderViolinPlot(graphElementId, results, {
         plotType: distributionPlotToUse,
         showPoints: distributionPoints
       })
+
+      const end = `perfTimeFrontendEnd-${graphElementId}`
+      performance.mark(end)
```

Comment:
same about mark(end), and perhaps moving these lines after into a logPerformance method so that the loadData() method reads cleaner

Reply:
Done in a517ef9.